doc/src/sgml/xlstproc_depfile_wrapper.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+import subprocess
+import sys
+import argparse
+from unittest import result
+
+parser = argparse.ArgumentParser(
+    description='generate dependency file for docs')
+
+parser.add_argument('--basename', type=str, required=False, nargs='+')
+parser.add_argument('--depfile', type=str, required=False)
+parser.add_argument('-i', '--input', type=str, nargs='+')
+parser.add_argument('-o', '--output', type=str, required=True)
```

Comment:
Looks like this script doesn't actually need to parse -i / -o, --path, and  could just hand them off to xsltproc without parsing here?

Reply:
I think we need to parse at least '-i' since input files should be at the end. Also, I believe parsing these flags makes script less error prune and could help in the future(if any change is needed).

Reply:
> I think we need to parse at least '-i' since input files should be at the end.

If we just append the passed-in flags to the end of the commandline that's not a problem.

>  Also, I believe parsing these flags makes script less error prune and could help in the future(if any change is needed).

I think it does the opposite, since it requires very lenient option parsing.